libs/agno/agno/tools/function.py

```diff
@@ -422,7 +425,42 @@ def _wrap_callable(func: Callable) -> Callable:
             return wrapped
 
     def process_schema_for_strict(self):
-        self.parameters["additionalProperties"] = False
+        """Process the schema to make it strict mode compliant."""
+        def make_nested_strict(schema):
+            """Recursively ensure all object schemas have additionalProperties: false"""
+            if not isinstance(schema, dict):
+                return schema
+            
+            # Make a copy to avoid modifying the original
+            result = schema.copy()
+            
+            # If this is an object schema, ensure additionalProperties: false
+            if (result.get("type") == "object" or "properties" in result):
+                result["additionalProperties"] = False
+            
+            # If schema has no type but has other schema properties, give it a type
```

Comment:
When would this happen?

Reply:
so for the knowledge filters schema-
```
{
  "properties": {
    "key": {"type": "string"},
    "value": {
      "title": "Value",
       ... <- open ai expects a type field here
    }
  },
  "required": ["key", "value"]
}
```
if we dont do it we get an error like
```
agno.exceptions.ModelProviderError: Invalid schema for function 'search_knowledge_base_with_agentic_filters': In context=('properties', 'filters', 'items', 'properties', 'value'), schema must have a 'type' key.
```